src/core/mapping/base_mapper.cc

```diff
@@ -217,11 +217,13 @@ void BaseMapper::slice_task(const Legion::Mapping::MapperContext ctx,
   if (task.sharding_space.exists())
     sharding_domain = runtime->get_index_space_domain(ctx, task.sharding_space);
 
-  auto lo = key_functor->project_point(sharding_domain.lo(), sharding_domain);
-  auto hi = key_functor->project_point(sharding_domain.hi(), sharding_domain);
+  auto lo                    = key_functor->project_point(sharding_domain.lo(), sharding_domain);
+  auto hi                    = key_functor->project_point(sharding_domain.hi(), sharding_domain);
+  uint32_t total_tasks_count = linearize(lo, hi, hi) + 1;
+
   for (Domain::DomainPointIterator itr(input.domain); itr; itr++) {
-    auto p   = key_functor->project_point(itr.p, sharding_domain);
-    auto idx = linearize(lo, hi, p);
+    auto p       = key_functor->project_point(itr.p, sharding_domain);
+    uint32_t idx = linearize(lo, hi, p) * local_range.total_proc_count() / total_tasks_count;
```

Comment:
You need to add the starting processor id here so `idx` denotes a global processor id. I think we should modify `LocalProcessorRange` so its `offset_` be the starting processor id.

Reply:
do you think using local processor ID here will result in incorrect logic ? We use local proc ID down this line anyway

Reply:
`idx` per your calculation is an offset from the starting global processor id. that's why you needed a modulo operator in the `LocalProcessorRange::operator[]`. if you don't add the tarting global processor id, then you code won't handle correctly the case where a processor range start from a non-zero global processor id.

Reply:
would it be correct to calculate starting processor id as `machine_desc.processor_range().low`?

Reply:
yes